src/main/java/com/google/devtools/build/lib/exec/SpawnStrategyRegistry.java

```diff
@@ -595,6 +631,51 @@ public String toString() {
     }
   }
 
+  private static class StrategyPlatformFilter {
+    private final StrategyMapper strategyMapper;
+    private final ImmutableListMultimap<Label, SpawnStrategy> platformToStrategies;
+
+    private StrategyPlatformFilter(
+      StrategyMapper strategyMapper,
+      ImmutableListMultimap<Label, SpawnStrategy> platformToStrategies) {
+      this.strategyMapper = strategyMapper;
+      this.platformToStrategies = platformToStrategies;
+    }
+
+    public <T extends SpawnStrategy> List<T> getStrategies(
+        Spawn spawn, List<T> candidateStrategies) {
+      var platformLabel = spawn.getExecutionPlatformLabel();
+      Preconditions.checkNotNull(platformLabel, "Attempting to spawn action without an execution platform.");
+
+      if (platformToStrategies.containsKey(platformLabel)) {
+        var allowedStrategies = platformToStrategies.get(platformLabel);
+        List<T> filteredStrategies = new ArrayList<>();
+        for (var strategy : candidateStrategies) {
+          if (allowedStrategies.contains(strategy)) {
```

Comment:
This is treating the `allowedStrategies` as a set, and ignoring the ordering.

a) I assumed that `--allowed_strategies_by_exec_platform=@platforms//host=sandbox,standalone` implies that sandbox is preferred over standalone: is that not the intended semantics?
b) Is something else (possibly the ordering in`candidateStrategies` enforcing that?

I am specifically interested in the case where the flags are `--spawn_strategy=standalone,sandbox --allowed_strategies_by_exec_platform=@platforms//host=sandbox,standalone` (ie, they are reversed between the flags). I assume that if the platform matches, the ordering from `--allowed_strategies_by_exec_platform` has precedence. Please confirm if this is intended, and make sure the code enforces it (and consider adding a test for this case).

Reply:
The intent was that `--allowed_strategies_by_exec_platform` only served to filter out disallowed strategies, preserving order of candidate strategies. That said this was not accounted for in the proposal, documentation nor covered by tests. Good catch!